File: src/classes/Party.java
package src.classes;
import java.util.LinkedList;


public class Party {
	// attributes
	private int number = 0;
	private String name = "";
	private String alias = "";
	private int partyVotes = 0;
	private int totalVotes = 0;
	private LinkedList<Candidate> candidates = null;


	// constructor method
	public Party(String alias) {
		this.alias = alias;
	}


	// methods
	public void displayPartyInfo() {
		System.out.println("Party alias: " + alias);
		System.out.println("Party number: " + number);
		System.out.println("Party name: " + name);
		System.out.println("Party votes: " + partyVotes);
		System.out.println("Total votes: " + totalVotes);
		System.out.println();
	}

	public int getNumber() {
		return number;
	}
	public void setNumber(int number) {
		this.number = number;
	}

	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}

	public String getAlias() {
		return alias;
	}
	public void setAlias(String alias) {
		this.alias = alias;
	}

	public int getPartyVotes() {
		return partyVotes;
	}
	public void setPartyVotes(int votes) {
		this.partyVotes = votes;
	}

	public int getTotalVotes() {
		return totalVotes;
	}
	public void setTotalVotes(int votes) {
		this.totalVotes = votes;
	}

	public LinkedList<Candidate> getCandidates() {
		return candidates;
	}
	public void setCandidates(LinkedList<Candidate> candidates) {
		this.candidates = candidates;
	}
}

File: src/party.cpp
#include "../include/party.h"


Party::Party(const string &alias) {
	this->alias = alias;
}


void Party::displayPartyInfo() {
	cout << "Party alias: " << alias << endl;
	cout << "Party number: " << number << endl;
	cout << "Party name: " << name << endl;
	cout << "Party votes: " << partyVotes << endl;
	cout << "Total votes: " << totalVotes << endl;
	cout << "\n";
}

const int Party::getNumber() {
	return number;
}
void Party::setNumber(const int number) {
	this->number = number;
}

const string& Party::getName() {
	return name;
}
void Party::setName(const string &name) {
	this->name = name;
}

const string& Party::getAlias() {
	return alias;
}
void Party::setAlias(const string &alias) {
	this->alias = alias;
}

const int Party::getPartyVotes() {
	return partyVotes;
}
void Party::setPartyVotes(const int votes) {
	this->partyVotes = votes;
}

const int Party::getTotalVotes() {
	return totalVotes;
}
void Party::setTotalVotes(const int votes) {
	this->totalVotes = votes;
}

vector<Candidate*> Party::getCandidates() {
	return candidates;
}
void Party::setCandidates(vector<Candidate*> candidates) {
	this->candidates = candidates;
}
void addCandidate(Candidate &candidate) {
	candidates.push_back(&candidate);
}
void removeCandidate(Candidate &candidate) {
	candidates.remove(&candidate);
}
